mailpoet/lib/Segments/WP.php

```
@@ -182,8 +182,24 @@ private function handleCreatingOrUpdatingSubscriber(string $currentFilter, \WP_U
     }
 
+    // When updating an existing subscriber's email, check if another subscriber
+    // already holds the new email to avoid a unique constraint violation.
+    // This can happen when a WP user registers with email A, checks out with
+    // email B (creating a second subscriber), then changes their account email
+    // from A to B.
+    // Both the duplicate removal and the subscriber update are wrapped in a
+    // transaction so the delete is rolled back if the update fails.
     try {
-      $subscriber = $this->createOrUpdateSubscriber($data, $subscriber);
+      $this->entityManager->wrapInTransaction(function () use (&$subscriber, $data) {
```

PHPStan (2.1.46)

[error] Call to method wrapInTransaction() on an unknown class MailPoetVendor\Doctrine\ORM\EntityManager.
Learn more at https://phpstan.org/user-guide/discovering-symbols

(class.notFound)